modules/runtime-args/src/main/native/android/c/runtimeargs.c

```diff
@@ -0,0 +1,100 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GLUON BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "runtimeargs.h"
+
+// Graal handles
+static jclass jGraalRuntimeArgsClass;
+static jmethodID jGraalRuntimeArgsProcessMethod;
+
+static void initializeGraalHandles(JNIEnv* env) {
+    jGraalRuntimeArgsClass = (*env)->NewGlobalRef(env, (*env)->FindClass(env, "com/gluonhq/attach/runtimeargs/impl/AndroidRuntimeArgsService"));
+    jGraalRuntimeArgsProcessMethod = (*env)->GetStaticMethodID(env, jGraalRuntimeArgsClass, "processRuntimeArgs", "(Ljava/lang/String;Ljava/lang/String;)V");
+}
+
+static void initializeRuntimeArgsDalvikHandles() {
+    jclass activityClass = substrateGetActivityClass();
+    jclass jRuntimeArgsServiceClass = substrateGetRuntimeArgsServiceClass();
+
+    ATTACH_DALVIK();
+    jmethodID jRuntimeArgsServiceInitMethod = (*dalvikEnv)->GetMethodID(dalvikEnv, jRuntimeArgsServiceClass, "<init>", "(Landroid/app/Activity;)V");
+    jthrowable t = (*dalvikEnv)->ExceptionOccurred(dalvikEnv);
+    if (t) {
+        ATTACH_LOG_INFO("EXCEPTION occurred when dealing with dalvik handles\n");
+        (*dalvikEnv)->ExceptionClear(dalvikEnv);
+    }
+
+    jobject jActivity = substrateGetActivity();
+    jobject jObj = (*dalvikEnv)->NewObject(dalvikEnv, jRuntimeArgsServiceClass, jRuntimeArgsServiceInitMethod, jActivity);
+    jobject jDalvikRuntimeArgsService = (jobject)(*dalvikEnv)->NewGlobalRef(dalvikEnv, jObj);
+    DETACH_DALVIK();
+}
+
+//////////////////////////
+// From Graal to native //
+//////////////////////////
+
+JNIEXPORT jint JNICALL
+JNI_OnLoad_runtimeargs(JavaVM *vm, void *reserved)
+{
+#ifdef JNI_VERSION_1_8
+    JNIEnv* graalEnv;
+    if ((*vm)->GetEnv(vm, (void **)&graalEnv, JNI_VERSION_1_8) != JNI_OK) {
+        ATTACH_LOG_WARNING("Error initializing native RuntimeArgs from OnLoad");
+        return JNI_FALSE;
+    }
+    ATTACH_LOG_FINE("Initializing native RuntimeArgs from OnLoad");
+    initializeGraalHandles(graalEnv);
+    initializeRuntimeArgsDalvikHandles();
+    ATTACH_LOG_FINE("Initializing native RuntimeArgs from OnLoad Done");
+    return JNI_VERSION_1_8;
+#else
+    #error Error: Java 8+ SDK is required to compile Attach
```

Comment:
Don't we require 11 or higher?

Reply:
This is the standard `JNI_OnLoad_$Service` we have for every Attach service, but yes, we require JDK 11+ to compile the Attach Java (GraalVM) classes (1.7 for the Dalvik ones).
 
Do you mean we have to change this to `JNI_VERSION_10`?